lib/timescaledb/schema_dumper.rb

```diff
@@ -4,16 +4,22 @@
 module Timescaledb
   module SchemaDumper
     def tables(stream)
+      # Create fake streams to capture the output for these objects
+      @hypertables = StringIO.new
+      @retention_policies = StringIO.new
+
       super # This will call #table for each table in the database
       views(stream) unless defined?(Scenic) # Don't call this twice if we're using Scenic
+      timescale_hypertables(stream)
+      timescale_retention_policies(stream)
     end
 
     def table(table_name, stream)
       super(table_name, stream)
       if Timescaledb::Hypertable.table_exists? &&
          (hypertable = Timescaledb::Hypertable.find_by(hypertable_name: table_name))
-        timescale_hypertable(hypertable, stream)
-        timescale_retention_policy(hypertable, stream)
+        timescale_hypertable(hypertable, @hypertables)
```

Comment:
I couldn't understand the benefits of using these extra streams here. Any special reason?

Reply:
Discussed in my other comment - but I wanted to avoid iterating over the tables multiple times.

If we were fine with iterating multiple times, then we could:

1. Drop the custom `table` method entirely.
2. Iterate in `timescale_hypertables` and `timescale_retention_policies`, and output directly to the main stream.

I wasn't sure which was cleaner 😁

Reply:
Let's just add the retention policies first, and then later we will work on this refactoring isolatedly.